src/workflows/client/client.py

```diff
@@ -0,0 +1,267 @@
+                    elif response.status_code == 204:
+                        # Handler completed, no more events
+                        return
+
+                    response.raise_for_status()
+
+                    async for line in response.aiter_lines():
+                        if line.strip():  # Skip empty lines
+                            try:
+                                event = json.loads(line.replace("\n", ""))
+                                yield event
+                            except json.JSONDecodeError as e:
+                                print(f"Failed to parse JSON: {e}, data: {line}")
+                                continue
+
+            except httpx.TimeoutException:
+                raise TimeoutError(
+                    f"Timeout waiting for events from handler {handler_id}"
+                )
+            except httpx.RequestError as e:
+                raise ConnectionError(f"Failed to connect to event stream: {e}")
+
+    async def send_event(
+        self,
+        handler_id: str,
+        event: Event | dict[str, Any] | str,
+        step: str | None = None,
+    ) -> bool:
+        """
+        Send an event to the workflow.
+
+        Args:
+            handler_id (str): ID of the handler of the running workflow to send the event to
+            event (Event | dict[str, Any] | str): Event to send, represented as an Event object, a dictionary or a serialized string.
+            step (str | None): Step to send the event to (optional, defaults to None)
+
+        Returns:
+            bool: Success status of the send operation
+        """
+        if isinstance(event, Event):
+            try:
+                event = event.model_dump_json()
+            except Exception as e:
+                raise ValueError(f"Error while serializing the provided event: {e}")
+        elif isinstance(event, dict):
+            try:
+                event = json.dumps(event)
+            except Exception as e:
+                raise ValueError(f"Error while serializing the provided event: {e}")
+        request_body = {"event": event}
+        if step:
+            request_body.update({"step": step})
+        async with self._get_client() as client:
+            response = await client.post(f"/events/{handler_id}", json=request_body)
+            response.raise_for_status()
+
+            return response.json()["status"] == "sent"
+
+    async def get_result(self, handler_id: str) -> Any:
+        """
+        Get the result of the workflow associated with the specified handler ID.
+
+        Args:
+            handler_id (str): ID of the handler running the workflow
+
+        Returns:
+            Any: Result of the workflow
+        """
+        async with self._get_client() as client:
+            response = await client.get(f"/results/{handler_id}")
+            response.raise_for_status()
+
+            if response.status_code == 202:
+                return
+
+            return response.json()["result"]
```

Comment:
I think we should expose the full response TypedDict (already defined for the server), and return that. That way you can get the current status, etc.